### PR TITLE
fix: [CI-3998]: persist false for automountServiceAccountToken

### PR DESCRIPTION
**Pull request by mtran7**
##### Screenshots:

image

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by sahithikolichala — APPROVED** [review]

**Comment by mtran7** [comment]
retigger titlecheck

**Comment by mtran7** [comment]
retrigger titlecheck

**Review by chiragp1212 — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Comment by mtran7** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx [hunk 1]
```diff
@@ -737,7 +735,7 @@ export function StageInputSetFormInternal({
                 labelKey: 'pipeline.buildInfra.addCapabilities',
                 tooltipId: 'addCapabilities'
               })}
-            {(deploymentStageTemplate.infrastructure as any).spec?.containerSecurityContext?.capabilities.drop &&
+            {(deploymentStageTemplate.infrastructure as any).spec?.containerSecurityContext?.capabilities?.drop &&
```
Comment (sahithikolichala) [inline src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:738]: can we have the right tupe instead of any

**Review by sahithikolichala — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx (hunk 1 above)
Comment (mtran7) [reply src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:738]: Created a follow up ticket for this https://harness.atlassian.net/browse/CI-4011

**Pull request merged**